internal/repository/order/repository.go

```diff
@@ -0,0 +1,263 @@
+			VALUES ($1, $2, $3) RETURNING "id", "user_uuid", "order_id", "status", "accrual", "uploaded_at", "processed_at"`
+
+	updateQuery = `
+		UPDATE orders 
+		SET 
+		    "status" = $1, "accrual" = $2, "processed_at" = NOW()
+		WHERE "order_id" = $3
+		RETURNING "id", "user_uuid", "order_id", "status", "accrual", "uploaded_at", "processed_at"`
+
+	selectByIDQuery = `
+		SELECT "id", "user_uuid", "order_id", "status", "accrual", "uploaded_at", "processed_at" FROM orders WHERE "order_id" = $1`
+
+	selectByIDsQueryList = `
+		SELECT "order_id" FROM orders WHERE "order_id" = ANY($1::varchar[]) AND "status" = ANY($2::order_status_type[])`
+
+	selectByUserAndIDQuery = `
+		SELECT "id", "user_uuid", "order_id", "status", "accrual", "uploaded_at", "processed_at" FROM orders WHERE "user_uuid" = $1 AND "order_id" = $2`
+
+	selectByUserQuery = `
+		SELECT "id", "user_uuid", "order_id", "status", "accrual", "uploaded_at", "processed_at" FROM orders WHERE "user_uuid" = $1 ORDER BY "uploaded_at" DESC`
+
+	selectByStatusQuery = `
+		SELECT "id", "user_uuid", "order_id", "status", "accrual", "uploaded_at", "processed_at" FROM orders WHERE "status" = ANY($1::order_status_type[]) ORDER BY "uploaded_at"`
+)
+
+// Repository Репозиторий заказов.
+type Repository struct {
+	db     *db.ConnWrapper
+	logger *slog.Logger
+}
+
+// Order Заказ по orderID.
+func (r *Repository) Order(ctx context.Context, orderID string) (*entity.Order, error) {
+	c, cancel := context.WithTimeout(ctx, db.RequestTimeout)
+	defer cancel()
+
+	var order entity.Order
+	err := r.fullScan(
+		r.db.Connection(ctx).QueryRow(c, selectByIDQuery, orderID),
+		&order,
+	)
+	if err != nil {
+		if errors.Is(err, pgx.ErrNoRows) {
+			return nil, nil
+		}
+
+		return nil, err
+	}
+
+	return &order, nil
+}
+
+// OrdersList Заказы по orderID.
+func (r *Repository) OrdersList(ctx context.Context, orderIDs []string, st ...entity.OrderStatus) ([]string, error) {
+	c, cancel := context.WithTimeout(ctx, db.RequestTimeout)
+	defer cancel()
+
+	statuses := *(*[]string)(unsafe.Pointer(&st))
+	rows, err := r.db.Connection(ctx).Query(
+		c,
+		selectByIDsQueryList,
+		"{"+strings.Join(orderIDs, ",")+"}",
+		"{"+strings.Join(statuses, ",")+"}",
+	)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	orders := make([]string, 0)
+	for rows.Next() {
+		var order string
+		if err = rows.Scan(&order); err != nil {
+			return nil, err
+		}
+		orders = append(orders, order)
+	}
+	if rows.Err() != nil {
+		return nil, rows.Err()
+	}
+
+	return orders, nil
+}
+
+// OrderByUser Заказ по userUUID и orderID.
+func (r *Repository) OrderByUser(ctx context.Context, userUUID, orderID string) (*entity.Order, error) {
+	c, cancel := context.WithTimeout(ctx, db.RequestTimeout)
+	defer cancel()
+
+	return r.queryRow(c, selectByUserAndIDQuery, userUUID, orderID)
+}
+
+// OrdersByUser Заказы по userUUID.
+func (r *Repository) OrdersByUser(ctx context.Context, userUUID string) (*[]entity.Order, error) {
```

Comment:
Нет смысла использовать указатель на массив.
```suggestion
func (r *Repository) OrdersByUser(ctx context.Context, userUUID string) ([]entity.Order, error) {
```